File: src/NeuralDamage.Domain/Bot.cs
namespace NeuralDamage.Domain;

public class Bot : BaseEntity
{
    public required string Name { get; set; }
    public required string ModelId { get; set; }
    public string SystemPrompt { get; set; } = string.Empty;
    public string? Personality { get; set; }
    public double Temperature { get; set; } = 0.7;
    public string? AvatarUrl { get; set; }
    public string? Aliases { get; set; }
    public bool IsActive { get; set; } = true;

    public required Guid CreatedById { get; init; }
    public User CreatedBy { get; set; } = null!;

    public ICollection<ChatMember> ChatMembers { get; set; } = [];
    public ICollection<Message> Messages { get; set; } = [];
    public ICollection<Reaction> Reactions { get; set; } = [];
}

File: backend/app/schemas/bot.py
from datetime import datetime
from pydantic import BaseModel


class BotCreate(BaseModel):
    name: str
    model_id: str
    system_prompt: str
    personality: str | None = None
    avatar_url: str | None = None
    temperature: float = 0.7


class BotUpdate(BaseModel):
    name: str | None = None
    model_id: str | None = None
    system_prompt: str | None = None
    personality: str | None = None
    avatar_url: str | None = None
    temperature: float | None = None


class BotOut(BaseModel):
    id: str
    name: str
    avatar_url: str | None = None
    model_id: str
    system_prompt: str
    personality: str | None = None
    temperature: float
    created_by: str
    created_at: datetime
    is_active: bool

    model_config = {"from_attributes": True}
